pkg/cli/admin/inspect/resource.go

```diff
@@ -275,13 +275,18 @@ func obtainRelatedObjects(obj *unstructured.Unstructured) ([]*configv1.ObjectRef
 	}
 
 	relatedObjs := []*configv1.ObjectReference{}
-	for _, relatedObj := range val {
+	for i, relatedObj := range val {
 		ref := &configv1.ObjectReference{}
 		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(relatedObj.(map[string]interface{}), ref); err != nil {
 			return nil, err
 		}
-		relatedObjs = append(relatedObjs, ref)
-		klog.V(1).Infof("    Found related object %q...\n", objectReferenceToString(ref))
+		refStr := objectReferenceToString(ref)
```

Comment:
Thanks for opening this PR. Currently if ref is not objectreference, it will be skipped and must-gather will continue for other resources. In the bug description, it says that `ConfigurationPolicy` should be collected but I'm not sure that this change will make `ConfigurationPolicy` is gathered.

Reply:
It looks like the `gatherMoreObjects` function would continue after trying with the empty `ObjectReference` in this situation, saving the error to report later. But, [in the gatherGenericObject function](https://github.com/openshift/oc/blob/d2f8bfe9c582228f7cffd03834f6872dffdde623/pkg/cli/admin/inspect/resource.go#L193), it returns that error before writing the data for the root object (the `ConfigurationPolicy`, in my case).

The change I proposed removes the error. It might be better to not return immediately in `gatherGenericObject`, but I am not sure what other kinds of errors might appear there, which *should* cause it return immediately.

I understand testing with other people's CRDs is hard, so I have run `oc adm inspect configurationpolicies.policy.open-cluster-management.io` using a build with this change, and it did successfully gather the `ConfigurationPolicy`, if that makes you feel more confident.

Reply:
Thanks, I'm leaning towards what you said;

> The change I proposed removes the error. It might be better to not return immediately in gatherGenericObject, but I am not sure what other kinds of errors might appear there, which should cause it return immediately.

We only want to write the yaml of this object, when it is relatedobjects are totally not related. Maybe as you said, we can write the yaml and after that return error. WDYT?

Reply:
Yeah. I thought writing the YAML and then handling multiple possible errors would be annoying. But it doesn't seem to bad actually. Maybe something like this?
```diff
diff --git a/pkg/cli/admin/inspect/resource.go b/pkg/cli/admin/inspect/resource.go
index 3e0f061d8..a9f89097a 100644
--- a/pkg/cli/admin/inspect/resource.go
+++ b/pkg/cli/admin/inspect/resource.go
@@ -1,6 +1,7 @@
 package inspect

 import (
+	stderrors "errors"
 	"fmt"
 	"os"
 	"path"
@@ -186,12 +187,12 @@ func toStructuredObject[T any, TList any](obj runtime.Object) (runtime.Object, e
 }

 func gatherGenericObject(context *resourceContext, info *resource.Info, o *InspectOptions) error {
+	var gatherErr error
+
 	unstr, ok := info.Object.(*unstructured.Unstructured)
 	if ok {
 		// obtain associated objects for the current resource
-		if err := gatherRelatedObjects(context, unstr, o); err != nil {
-			return err
-		}
+		gatherErr = gatherRelatedObjects(context, unstr, o)
 	}

 	// save the current object to disk
@@ -199,10 +200,12 @@ func gatherGenericObject(context *resourceContext, info *resource.Info, o *Inspe
 	filename := filenameForInfo(info)
 	// ensure destination path exists
 	if err := os.MkdirAll(dirPath, os.ModePerm); err != nil {
-		return err
+		return stderrors.Join(err, gatherErr)
 	}

-	return o.fileWriter.WriteFromResource(path.Join(dirPath, filename), info.Object)
+	writeErr := o.fileWriter.WriteFromResource(path.Join(dirPath, filename), info.Object)
+
+	return stderrors.Join(writeErr, gatherErr)
 }

 func gatherRelatedObjects(context *resourceContext, unstr *unstructured.Unstructured, o *InspectOptions) error {
```

Reply:
Sounds reasonable. Could you please use error aggregator like in https://github.com/openshift/oc/blob/fe0d9d929496b6dfe8c4fb931ca7954f122b179c/pkg/cli/admin/inspect/resource.go#L65 instead of `stderrors`?